packages/mcp/src/server/tools/token-metadata-service.ts

```diff
@@ -0,0 +1,161 @@
+import * as fs from "fs";
+import * as path from "path";
+
+export interface TokenMetadata {
+  name: string;
+  value: string;
+  category: string;
+  description?: string;
+  file: string;
+}
+
+export class TokenMetadataService {
+  private static tokenCache: TokenMetadata[] | null = null;
+
+  /**
+   * Get all design tokens from the style package
+   */
+  static async getTokenMetadata(): Promise<TokenMetadata[]> {
+    if (this.tokenCache) {
+      return this.tokenCache;
+    }
+
+    this.tokenCache = await this.parseAllTokenFiles();
+    return this.tokenCache;
+  }
+
+  /**
+   * Parse all SCSS files in the style package to extract tokens
+   */
+  private static async parseAllTokenFiles(): Promise<TokenMetadata[]> {
+    const allTokens: TokenMetadata[] = [];
+
+    try {
+      const stylePackagePath = this.getStylePackagePath();
+      const srcPath = path.join(stylePackagePath, "src");
+
+      // Parse base token files
+      const tokenFiles = [
+        { file: "spacing.scss", category: "Spacing" },
+        { file: "typography.scss", category: "Typography" },
+        { file: "border-radius.scss", category: "Border Radius" },
+        { file: "borders.scss", category: "Borders" },
+        { file: "motion.scss", category: "Motion" }
+      ];
```

Comment:
Can we somehow add test or automate this in case filenames change/move? So it will not be broken